pkg/resource/managed_reconciler.go

```diff
@@ -235,17 +278,16 @@ type mrManaged struct {
 	ManagedFinalizer
 	ManagedInitializer
 	ManagedReferenceResolver
+	ManagedAnnotator
 }
 
 func defaultMRManaged(m manager.Manager) mrManaged {
 	return mrManaged{
 		ManagedConnectionPublisher: NewAPISecretPublisher(m.GetClient(), m.GetScheme()),
 		ManagedFinalizer:           NewAPIManagedFinalizerRemover(m.GetClient()),
```

Comment:
It seems #51 is not fixed in this PR.

Reply:
This PR does not literally make the finalizer mandatory, but it does return it to being on by default and less easy to accidentally turn off. The spirit of #51 seemed to be that it had become easy to accidentally turn off finalizers unintentionally, and I think that's no longer the case with this PR.